### DISCUSSION TITLE
[Feature Request] Support for gh .chart DoubleGuitar, DoubleRhythm, DoubleBass

### DISCUSSION THREAD

**Discussion by shockdude** [TeamRizu/OutFox#533] [03 - Feature Requests]
### Please Select the game mode your feature request is about.

gh

### Is your feature request related to a problem?

The gh .chart parser can only parse gh-solo ("Single") and gh-bass (the first "Double" chart that appears in the file, regardless if it's DoubleGuitar, DoubleBass, or DoubleRhythm).
It cannot parse DoubleGuitar into gh-solo, or DoubleRhythm into gh-rhythm.

### Describe the solution you'd like

The .chart parser should parse all guitar instruments available in .chart for gh:

- Single --> gh-solo
- DoubleGuitar --> gh-solo
- DoubleBass --> gh-bass
- DoubleRhythm --> gh-rhythm

There ~~may be rare cases~~ exist cases where a .chart contains both valid Single and DoubleGuitar charts, e.g. Soulless 4. For such cases, a dedicated "gh-coop" style for DoubleGuitar might be a good idea (and the .mid parser updated to support it too).

### Additional context

Support for these .chart styles would make it easier to chart for Project Serenity, specifically "CRUSH THE DEVIL"

**Comment by shockdude** [comment]
Test .chart with DoubleGuitar, DoubleRhythm, DoubleBass styles
[GH Double Style Test.zip](https://github.com/TeamRizu/OutFox/files/8440161/GH.Double.Style.Test.zip)

Test .chart with Single, DoubleGuitar, DoubleRhythm, DoubleBass styles
[GH Single & Double Style Test.zip](https://github.com/TeamRizu/OutFox/files/8440162/GH.Single.Double.Style.Test.zip)

**Comment by TheNathannator** [comment]
+1 for putting Co-op Guitar on its own style so it can be distinguished from the Lead Guitar part, I would not recommend putting it on gh-solo.

This is more on the pedantic side, but I'd also like to note that the phrasing here kinda makes it sound like having both a Single and DoubleGuitar track in the same .chart is more of an edge case. In reality, nothing is stopping new charts from being made that way, and the other major supporters of the .chart format (Clone Hero and Moonscraper) support this.
